polyfill/lib/calendar.mjs

```diff
@@ -333,53 +332,1354 @@ function resolveNonLunisolarMonth(calendarDate) {
+    const { year, month, day } = calendarDate;
+    const addedMonths = this.addMonthsCalendar({ year: year + years, month, day }, months, overflow, cache);
+    days += weeks * 7;
+    const addedDays = this.addDaysCalendar(addedMonths, days, cache);
+    return addedDays;
+  },
+  untilCalendar(calendarOne, calendarTwo, largestUnit, cache) {
+    let days = 0;
+    let weeks = 0;
+    let months = 0;
+    let years = 0;
+    switch (largestUnit) {
+      case 'days':
+        days = this.calendarDaysUntil(calendarOne, calendarTwo, cache);
+        break;
+      case 'weeks': {
+        const totalDays = this.calendarDaysUntil(calendarOne, calendarTwo, cache);
+        days = totalDays % 7;
+        weeks = (totalDays - days) / 7;
+        break;
+      }
+      case 'months':
+      case 'years': {
+        const diffYears = calendarTwo.year - calendarOne.year;
+        const diffMonths = calendarTwo.month - calendarOne.month;
+        const diffDays = calendarTwo.day - calendarOne.day;
+        const sign = this.compareCalendarDates(calendarTwo, calendarOne);
+        if (largestUnit === 'years' && diffYears) {
+          const isOneFurtherInYear = diffMonths * sign < 0 || (diffMonths === 0 && diffDays * sign < 0);
+          years = isOneFurtherInYear ? diffYears - sign : diffYears;
+        }
+        const yearsAdded = years ? this.addCalendar(calendarOne, { years }, 'constrain', cache) : calendarOne;
+        // Now we have less than one year remaining. Add one month at a time
+        // until we go over the target, then back up one month and calculate
+        // remaining days and weeks.
+        let current;
+        let next = yearsAdded;
+        do {
+          months++;
+          current = next;
+          next = this.addMonthsCalendar(current, sign, 'constrain', cache);
+          if (next.day !== calendarOne.day) {
+            // In case the day was constrained down, try to un-constrain it
+            next = this.regulateDate({ ...next, day: calendarOne.day }, 'constrain', cache);
+          }
+        } while (this.compareCalendarDates(calendarTwo, next) * sign >= 0);
+        months--; // correct for loop above which overshoots by 1
+        const remainingDays = this.calendarDaysUntil(current, calendarTwo, cache);
+        days = remainingDays % 7;
+        weeks = (remainingDays - days) / 7;
+        break;
+      }
     }
+    return { years, months, weeks, days };
+  },
+  daysInMonth(calendarDate, cache) {
+    // Add the length of the smallest possible month. If it rolls over to the
+    // next month, then back up to the end of the previous month to know its day
+    // count. If it doesn't roll over, add the same number of days again until
+    // we get a rollover.
+    //
+    // NOTE: This won't work for months where days are skipped (e.g. Gregorian
+    // switchover) which is OK because this is a prototype implementation.
```

Comment:
If the answer to my earlier question about this is 'unimportant for all ICU calendars', then could change language to 'which is OK because this does not affect any of the ICU calendars'

Reply:
Yep, I updated the comment.